From a6327db068a628a36c798e39c58d8e299c5eab4a Mon Sep 17 00:00:00 2001
From: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
Date: Thu, 12 May 2022 01:15:14 -0500
Subject: [PATCH] dump current boot post code to text file

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
---
 src/post_code.cpp | 22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/post_code.cpp b/src/post_code.cpp
index aa5aa44..4006297 100644
--- a/src/post_code.cpp
+++ b/src/post_code.cpp
@@ -17,6 +17,13 @@

 #include "iomanip"
 #include <lcdlib_common.h>
+#include <iostream>
+#include <fstream>
+#include <filesystem>
+
+#define POST_TXT_PATH "/var/lib/phosphor-post-code-manager/host0"
+#define PS_STR_BUFF_SIZE 128
+#define POST_TXT_FILE "-hex.txt"

 PostCodeDataHolder* PostCodeDataHolder::instance = 0;

@@ -110,11 +117,17 @@ void PostCode::savePostCodes(postcode_t code)
                         std::chrono::system_clock::now().time_since_epoch())
                         .count();

+    char post_file[PS_STR_BUFF_SIZE] = {0};
+    char ts_post[PS_STR_BUFF_SIZE] = {0};
+
     if (postCodes.empty())
     {
         firstPostCodeTimeSteady = postCodeTimeSteady;
         firstPostCodeUsSinceEpoch = tsUS; // uS since epoch for 1st post code
         incrBootCycle();
+        // remove if already exists, we've rotated boot counts
+        sprintf(post_file,"%s/%d%s",POST_TXT_PATH, currentBootCycleCount(), POST_TXT_FILE);
+        std::filesystem::remove(post_file);
     }
     else
     {
@@ -129,6 +142,15 @@ void PostCode::savePostCodes(postcode_t code)

     writeLCD(code);

+    // save latest LCD post code to text file
+    // for debugging current boot remotely - OPENBMC-632
+    std::ofstream post_text;
+    sprintf(post_file,"%s/%d%s",POST_TXT_PATH, currentBootCycleCount(), POST_TXT_FILE);
+    post_text.open (post_file, std::ios::out | std::ios::app);
+    sprintf(ts_post, "%lld : %llx\n", tsUS, std::get<uint64_t>(code));
+    post_text << ts_post;
+    post_text.close();
+
 #ifdef ENABLE_BIOS_POST_CODE_LOG
     std::ostringstream hexCode;
     hexCode << "0x" << std::setfill('0') << std::setw(2) << std::hex
