From 079484ed2b792853fe1fc746915a3fd13659c88c Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 2 Aug 2021 19:04:08 -0500
Subject: [PATCH 1/1] u-boot-sdk: eeprom board_id offset change

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 common/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/common/main.c b/common/main.c
index 0d3d9fb0d9..80226584aa 100644
--- a/common/main.c
+++ b/common/main.c
@@ -16,7 +16,7 @@

 #define EEPROM_I2C_BUS 7
 #define EEPROM_DEV_ADDR 0X50
-#define EEPROM_BOARD_ID_OFFSET 0X1F
+#define EEPROM_BOARD_ID_OFFSET 0X10
 #define EEPROM_ADDR_LEN 2
 #define BOARD_ID_LEN 1

--
2.17.1

