From 4d775c17a5365990f87a0070445cdca4a6efcd0d Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Wed, 12 May 2021 14:55:44 -0500
Subject: [PATCH] obmc-ikvm: kvm server set to listen on any interface

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 ikvm_server.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ikvm_server.cpp b/ikvm_server.cpp
index ebeaef0..01459d2 100644
--- a/ikvm_server.cpp
+++ b/ikvm_server.cpp
@@ -16,7 +16,7 @@ using namespace sdbusplus::xyz::openbmc_project::Common::Error;
 Server::Server(const Args& args, Input& i, Video& v) :
     pendingResize(false), frameCounter(0), numClients(0), input(i), video(v)
 {
-    std::string ip("localhost");
+    std::string ip("any");
     const Args::CommandLine& commandLine = args.getCommandLine();
     int argc = commandLine.argc;

--
2.17.1

