From b6d0abbfc67b34d2e82085bc9f209a2db5047f2c Mon Sep 17 00:00:00 2001
From: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
Date: Tue, 4 Jan 2022 06:14:47 -0600
Subject: [PATCH] add UDMA engine node to aspeed-g6

---
 arch/arm/boot/dts/aspeed-g6.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed-g6.dtsi
index f99d03d52f4e..f7f36098a9d1 100644
--- a/arch/arm/boot/dts/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed-g6.dtsi
@@ -740,6 +740,12 @@
 				status = "disabled";
 			};

+			udma: uart-dma@1e79e000 {
+				compatible = "aspeed,ast2600-udma";
+				reg = <0x1e79e000 0x400>;
+				interrupts = <GIC_SPI 56 IRQ_TYPE_LEVEL_HIGH>;
+			};
+
 			i3c: bus@1e7a0000 {
 				compatible = "simple-bus";
 				#address-cells = <1>;
--
2.17.1

