From 72d20fe64ff3637160ed95553a9bb39753a5fc96 Mon Sep 17 00:00:00 2001
From: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
Date: Thu, 16 Dec 2021 08:57:33 -0600
Subject: [PATCH] set BIOS boot type to EFI

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
---
 chassishandler.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/chassishandler.cpp b/chassishandler.cpp
index 4a3252a..1801c68 100644
--- a/chassishandler.cpp
+++ b/chassishandler.cpp
@@ -40,8 +40,8 @@ std::unique_ptr<phosphor::Timer> identifyTimer
 static ChassisIDState chassisIDState = ChassisIDState::reserved;
 static constexpr uint8_t setParmVersion = 0x01;
 static constexpr uint8_t setParmBootFlagsPermanent = 0x40;
-static constexpr uint8_t setParmBootFlagsValidOneTime = 0x80;
-static constexpr uint8_t setParmBootFlagsValidPermanent = 0xC0;
+static constexpr uint8_t setParmBootFlagsValidOneTime = 0xA0;
+static constexpr uint8_t setParmBootFlagsValidPermanent = 0xE0;

 constexpr size_t sizeVersion = 2;
 constexpr size_t DEFAULT_IDENTIFY_TIME_OUT = 15;
