From 65157a9a712ce4ee788ae70456ed6d123fc1277f Mon Sep 17 00:00:00 2001
From: Abinaya <abinaya.dhandapani@amd.com>
Date: Fri, 12 Aug 2022 01:43:29 -0500
Subject: [PATCH] uboot-aspeed-sdk:Add dtb selection for Shale

---
 common/image-fit.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/common/image-fit.c b/common/image-fit.c
index b23c0f45ec..5fdb239eb1 100644
--- a/common/image-fit.c
+++ b/common/image-fit.c
@@ -1871,9 +1871,8 @@ int fit_image_load(bootm_headers_t *images, ulong addr,
 					sprintf((char *)fit_uname_config, "conf-aspeed-bmc-amd-sunstone.dtb"); break;
 				case 0x62:
 				case 0x65:
-					sprintf((char *)fit_uname_config, "conf-aspeed-bmc-amd-shale96.dtb"); break;
 				case 0x59:
-					sprintf((char *)fit_uname_config, "conf-aspeed-bmc-amd-shale64.dtb"); break;
+					sprintf((char *)fit_uname_config, "conf-aspeed-bmc-amd-shale.dtb"); break;
 				case 0x63:
 					sprintf((char *)fit_uname_config, "conf-aspeed-bmc-amd-cinnabar.dtb"); break;
 				default :
--
2.25.1

