From ec4d5e5c119a9b4753fee7eed86602ace9535dee Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Mon, 14 Jun 2021 12:01:18 -0500
Subject: [PATCH 3/4] net eth mac address reading from eeprom
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 net/eth-uclass.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

diff --git a/net/eth-uclass.c b/net/eth-uclass.c
index a7f8792710..98d09c628f 100644
--- a/net/eth-uclass.c
+++ b/net/eth-uclass.c
@@ -12,9 +12,12 @@
 #include <dm/device-internal.h>
 #include <dm/uclass-internal.h>
 #include "eth_internal.h"
+#include <i2c.h>

 DECLARE_GLOBAL_DATA_PTR;

+#define I2C_MAC_OFFSET 0x8
+
 /**
  * struct eth_device_priv - private structure for each Ethernet device
  *
@@ -465,6 +468,7 @@ static int eth_post_probe(struct udevice *dev)
 	struct eth_device_priv *priv = dev->uclass_priv;
 	struct eth_pdata *pdata = dev->platdata;
 	unsigned char env_enetaddr[ARP_HLEN];
+	int ret;

 #if defined(CONFIG_NEEDS_MANUAL_RELOC)
 	struct eth_ops *ops = eth_get_ops(dev);
@@ -498,7 +502,21 @@ static int eth_post_probe(struct udevice *dev)
 	if (eth_get_ops(dev)->read_rom_hwaddr)
 		eth_get_ops(dev)->read_rom_hwaddr(dev);

-	eth_env_get_enetaddr_by_index("eth", dev->seq, env_enetaddr);
+	if (!eth_env_get_enetaddr_by_index("eth", dev->seq, env_enetaddr)) {
+		i2c_set_bus_num(CONFIG_SYS_I2C_EEPROM_BUS);
+		ret = i2c_read(CONFIG_SYS_I2C_EEPROM_ADDR, I2C_MAC_OFFSET, CONFIG_SYS_I2C_EEPROM_ADDR_LEN, env_enetaddr, ARP_HLEN);
+		if (ret < 0) {
+			printf("Error : Retrieving MAC from EEPROM Failed\n");
+		}
+		else {
+			if (is_valid_ethaddr(env_enetaddr)) {
+				eth_env_set_enetaddr_by_index("eth", dev->seq, env_enetaddr);
+				/* Once we have board ID env save working, we might not need to save here */
+				env_save();
+			}
+		}
+	}
+
 	if (!is_zero_ethaddr(env_enetaddr)) {
 		if (!is_zero_ethaddr(pdata->enetaddr) &&
 		    memcmp(pdata->enetaddr, env_enetaddr, ARP_HLEN)) {
@@ -512,6 +530,7 @@ static int eth_post_probe(struct udevice *dev)

 		/* Override the ROM MAC address */
 		memcpy(pdata->enetaddr, env_enetaddr, ARP_HLEN);
+		printf("Using ENV MAC address: %pM\n", env_enetaddr);
 	} else if (is_valid_ethaddr(pdata->enetaddr)) {
 		eth_env_set_enetaddr_by_index("eth", dev->seq, pdata->enetaddr);
 		printf("\nWarning: %s using MAC address from ROM\n",
--
2.17.1

