From 4faf4db332fc94bdb150f4ec227c23ece53a86bd Mon Sep 17 00:00:00 2001
From: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
Date: Fri, 26 Nov 2021 06:28:27 -0600
Subject: [PATCH] fix str2hex function

---
 common/image-fit.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/common/image-fit.c b/common/image-fit.c
index e2000c5ea1..3bdcc86130 100644
--- a/common/image-fit.c
+++ b/common/image-fit.c
@@ -1790,17 +1790,18 @@ static const char *fit_get_image_type_property(int type)
 #define HEX_BASE	16
 #define DECIMAL_BASE	10

-uint32_t str2hex(char *str)
+uint32_t str2hex(const char *str)
 {
 	uint32_t val = 0;
-
-	for (int i = 0; i < strlen(str); i++) {
-		if (isxdigit(str[i])) {
-			if (isdigit(str[i])) str[i] = str[i] - '0';
-			else str[i] = toupper(str[i]) - 'A'+ DECIMAL_BASE;
+	char *str1 = strdup(str);
+	for (int i = 0; i < strlen(str1); i++) {
+		if (isxdigit(str1[i])) {
+			if (isdigit(str1[i])) str1[i] = str1[i] - '0';
+			else str1[i] = toupper(str1[i]) - 'A'+ DECIMAL_BASE;
 		}
-		val = (val * HEX_BASE) | (str[i] & 0xF);
+		val = (val * HEX_BASE) | (str1[i] & 0xF);
 	}
+	free(str1);
 	return val;
 }

