From fd7c3da59279fb5dc91e665c384ec013fec8258a Mon Sep 17 00:00:00 2001
From: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
Date: Fri, 11 Feb 2022 06:17:27 -0600
Subject: [PATCH] read HPM FPGA version from FPGA i2c slave

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>
---
 item_updater.cpp | 19 ++++++++++++++++++-
 1 file changed, 18 insertions(+), 1 deletion(-)

diff --git a/item_updater.cpp b/item_updater.cpp
index b00dbe0..b86ea7a 100644
--- a/item_updater.cpp
+++ b/item_updater.cpp
@@ -37,6 +37,8 @@
         } \
         is.close(); \
     }
+#define PLAT_GET_HPM_FPGA_VER    ("amd-plat-info get_hpm_fpga_ver")
+#define BUFFLEN                  (64)

 namespace phosphor
 {
@@ -855,6 +857,9 @@ void ItemUpdater::createSCMFPGAObject()
 void ItemUpdater::createHPMFPGAObject()
 {
     std::string path = HPM_FPGA_OBJPATH;
+    FILE *pf;
+    char buff[BUFFLEN] = {0};
+
     // Get version id from last item in the path
     auto pos = path.rfind("/");
     if (pos == std::string::npos)
@@ -869,7 +874,19 @@ void ItemUpdater::createHPMFPGAObject()

     auto versionId = path.substr(pos + 1);
     std::string version = "Unknown";
-    GET_VER(HPM_VER_FILE);
+    /* read from FPGA slave first */
+    pf = popen(PLAT_GET_HPM_FPGA_VER, "r");
+    if (pf)
+    {
+        if (fgets(buff, BUFFLEN, pf) != NULL)
+            version = std::string(buff);
+        fclose(pf);
+    }
+    /* read from persistent store if slave returned unknown */
+    log<level::ERR>("version from slave ",
+                   entry("v=%s", version.c_str()));
+    if (version.compare("Unknown") == 0)
+        GET_VER(HPM_VER_FILE);

     AssociationList assocs = {};
     hpmfpgaActivation = std::make_unique<Activation>(
